vip-feed-cache.php

```diff
@@ -0,0 +1,99 @@
+<?php
+// Require core feed class.
+require_once( ABSPATH . WPINC . '/class-feed.php' );
+
+class VIPGO_Feed_Cache extends SimplePie_Cache {
+	/**
+	 * Creates a new SimplePie_Cache object.
+	 *
+	 * @access public
+	 *
+	 * @param string $location  URL location (scheme is used to determine handler).
+	 * @param string $filename  Unique identifier for cache object.
+	 * @param string $extension 'spi' or 'spc'.
+	 * @return VIPGO_Feed_Cache_Transient Feed cache handler object that uses transients and normalizes SimplePie Build number.
+	 */
+	public function create( $location, $filename, $extension ) {
+		return new VIPGO_Feed_Cache_Transient( $location, $filename, $extension );
+	}
+}
+
+class VIPGO_Feed_Cache_Transient extends WP_Feed_Cache_Transient {
+	/**
+	 * Constructor.
+	 *
+	 * @access public
+	 *
+	 * @param string $location  URL location (scheme is used to determine handler).
+	 * @param string $filename  Unique identifier for cache object.
+	 * @param string $extension 'spi' or 'spc'.
+	 */
+	public function __construct($location, $filename, $extension) {
+		parent::__construct( $location, $filename, $extension );
+	}
+
+	/**
+	 * Gets the transient.
+	 *
+	 * This also normalizes the SimplePie Build number.  If the returned build
+	 * number differes from what is expected, the cache is considered invalid.
+	 * The number can differ if one web container sets the cache and a different
+	 * web container reads the cache.  This is because the build number is set
+	 * by the filemtime() of the SimplePie source files, and file modified dates
+	 * can differ between web containers.
+	 *
+	 * @access public
+	 *
+	 * @return mixed Transient value.
+	 */
+	public function load() {
+		$transient = get_transient( $this->name );
+
+		// If we don't have the required data, bail.
+		if ( ! defined( 'SIMPLEPIE_BUILD' ) || ! isset( $transient['build'] ) ) {
+			return $transient;
+		}
+		$transient['build'] = SIMPLEPIE_BUILD;
```

Comment:
Unfortunately not.  Because of how SimplePie checks the `SIMLPEPIE_BUILD` it is outside of our _non-core-hack_ control.  The best we can do is ignore whatever it's set to and fool SimplePie into thinking it's correct.  As long as we use object caching for transients and the cache is cleared on core WordPress updates, there shouldn't be any issue with incompatible cache data.